### ISSUE TITLE
Issue with the RegExp replacement of the searchTerm

### ISSUE THREAD

**Issue by Katulka** [pozil/sfdc-ui-lookup-lwc#37]
Only in Production, having an issue with this line:

result.titleFormatted = result.title
                    ? result.title.replace(regex, '<b>' + this.searchTerm + '</b>')
                    : result.title;

this.searchTerm = "he"
result.title = "obfuscated community name he"

Testing:
- Only happens on the first page load
- Get failure on that line after user types "he" in the lookup
- Error is blank in the "catch" of the js
- Type he again and it works fine

Cannot replicate this in sandbox.

For now I am getting rid of RegExp and doing this:

result.title.replace(this.searchTerm, '<b>' + this.searchTerm + '</b>')

**Comment by pozil** [comment]
Hi @Katulka, if I get this right, you only have an issue on the JS side in production, right?
Could there there's something special in your prod data?

Can you check that by logging search results in the dev console just before applying the regex like this:
```js
this.searchResults = results.map((result) => {
   console.log(JSON.stringify(result, null, 2));
   ...
```

**Comment by Katulka** [comment]
Hi @pozil thanks so much.  Hope you are well.

FYI - this is helping a community screen visitors for Covid-19 reasons.. so it's important right now. Thanks for helping.

Here you go.. I am also finding it's not the RegExp but simply the assignment of result.titleFormatted


{
  "icon": "utility:world",
  "id": "a0C0B00000u6666UAA",
  "sObjectType": "Community",
  "subtitle": "Community • obfuscated community name",
  "title": "obfuscated community name"
}
modules/c/lookup.js:4 {
  "icon": "utility:world",
  "id": "a0C0B00000u6666UA2",
  "sObjectType": "Community",
  "subtitle": "Community • obfuscated community name",
  "title": "obfuscated community name"
}
modules/c/lookup.js:4 {
  "icon": "utility:world",
  "id": "a0C0B00000u6666UA2",
  "sObjectType": "Community",
  "subtitle": "Community • obfuscated community name",
  "title": "obfuscated community name"
}
modules/c/lookup.js:4 {
  "icon": "utility:world",
  "id": "a0CU00000006666MAI",
  "sObjectType": "Community",
  "subtitle": "Community • obfuscated community name",
  "title": "obfuscated community name"
}
modules/c/lookup.js:4 {
  "icon": "utility:world",
  "id": "a0C0B00000u6666UAU",
  "sObjectType": "Community",
  "subtitle": "Community • obfuscated community name",
  "title": "obfuscated community name"
}
modules/c/lookup.js:4 {
  "icon": "utility:world",
  "id": "a0CU000000b6666MA4",
  "sObjectType": "Community",
  "subtitle": "Community • obfuscated community name",
  "title": "obfuscated community name"
}
modules/c/lookup.js:4 {
  "icon": "utility:world",
  "id": "a0CU000000b6666MA4",
  "sObjectType": "Community",
  "subtitle": "Community • obfuscated community name",
  "title": "obfuscated community name"
}
modules/c/lookup.js:4 {
  "icon": "utility:world",
  "id": "a0C0B00000s6666UAS",
  "sObjectType": "Community",
  "subtitle": "Community • obfuscated community name",
  "title": "obfuscated community name"
}

**Comment by Katulka** [comment]
This just fixed it, I think it's a scope/lwc restriction.. cut ties with the reference and it's working

 setSearchResults(results) {
        // Reset the spinner
        this.loading = false;

        let resultsLocal = JSON.parse(JSON.stringify(results));

        this.searchResults = resultsLocal.map((result) => {

**Comment by Katulka** [comment]
![image](https://user-images.githubusercontent.com/6976559/79007859-1a66a980-7b2a-11ea-829e-6f88379a87fb.png)

Working version, showing results on first attempt.  This is a great lookup thanks!

**Comment by pozil** [comment]
Thanks for finding the issue so quickly @Katulka.
I've released v1.3.2 with the fix.

Good luck with your project and stay safe.

**Comment by Katulka** [comment]
You are welcome and thanks @pozil
